### PR TITLE
feat: log edge errors to datadog

### LINKED ISSUE #178: PyPI Edge Errors -> Datadog logs
@miketheman requested that we have edge errors for the PyPI service report to Datadog logs while debugging NGWAF errors recently.

This should be straightforward using similar configuration (and the existing secret for the api token) that we use for the files service [here](https://github.com/pypi/infra/blob/88ceeafdc5b67c9c4e90881376742559a7fcc1a7/terraform/file-hosting/fastly-service.tf#L166-L178).

### PR DESCRIPTION
**Pull request by JacobCoffee**
Closes #178

### CODE REVIEW

File: terraform/warehouse/main.tf [hunk 1]
```diff
@@ -194,6 +195,13 @@ resource "fastly_service_vcl" "pypi" {
     placement      = "none"
   }
 
+  logging_datadog {
+    name               = "Log Edge Errors"
+    token              = var.datadog_token
+    response_condition = "Edge Error"
+    format             = "{ \"ddsource\": \"fastly\", \"service\": \"%%{req.service_id}V\", \"date\": \"%%{begin:%Y-%m-%dT%H:%M:%S%z}t\", \"url\": \"%%{json.escape(req.url)}V\", \"message\": \"Edge error occurred!\", \"short_message\": \"edge_error\" }"
```
Comment (ewdurbin) [inline terraform/warehouse/main.tf:202]: @miketheman do you have input on what format would be most usable here?

**Review by ewdurbin — COMMENTED** [review]
Looks like the terraform plan is failing, and I'm curious if we want a bit more information in the log sent to DD.

File: terraform/warehouse/main.tf (hunk 1 above)
Comment (miketheman) [reply terraform/warehouse/main.tf:202]: [Datadog recommends](https://docs.datadoghq.com/integrations/fastly/) using the `Datadog-Fastly` format - https://docs.fastly.com/en/guides/log-streaming-datadog#using-the-json-logging-format (reminder it has to be made a single line and escaped).
Any reason to _not_ use their recommendation?

Comment (JacobCoffee) [reply terraform/warehouse/main.tf]: wdyt [efe3d73](https://github.com/pypi/infra/pull/179/commits/efe3d73ea0048c9e4db9497f12c589477476d1d3)?

I dont see any reason not to except it's a lot longer :)

Comment (miketheman) [reply terraform/warehouse/main.tf]: Seems like a reasonable (verbose, but hey, it's an error!) log format, and we can add any keys as needed once deployed. Yay JSON!

Comment (ewdurbin) [reply terraform/warehouse/main.tf]: god that is horrible to look at :shipit:

**Review by ewdurbin — APPROVED** [review]

**Comment by JacobCoffee** [comment]
This is live but it isn't working

**Pull request merged**